rust/ql/test/library-tests/type-inference/main.rs

```diff
@@ -1,4 +1,69 @@
-mod m1 {
+mod field_access {
+    #[derive(Debug)]
+    struct S;
+
+    #[derive(Debug)]
+    struct MyThing {
+        a: S,
+    }
+
+    #[derive(Debug)]
+    enum MyOption<T> {
+        MyNone(),
+        MySome(T),
+    }
+
+    #[derive(Debug)]
+    struct GenericThing<A> {
+        a: A,
+    }
+
+    struct OptionS {
+        a: MyOption<S>,
+    }
+
+    fn simple_field_access() {
+        let x = MyThing { a: S };
+        println!("{:?}", x.a);
+    }
+
+    fn generic_field_access() {
```

Comment:
Do all the expressions in this test have correctly inferred types (I didn't bother to check the `.expected` file)?

Reply:
I believe so, yes :)